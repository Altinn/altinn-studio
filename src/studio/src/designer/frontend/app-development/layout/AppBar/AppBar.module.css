.root {
  flex-grow: 1;
  z-index: 1;
  color: #000;
}

.aImgStyling {
  border-bottom: 'none';
}
.aImgStyling:hover {
  border-bottom: 'none';
}
.aImgStyling:active {
  border-bottom: 'none';
}

.appBar {
  border-bottom: '1px solid';
  border-bottom-color: '#C9C9C9';
  color: #000;
  position: 'fixed';
}

.appBarDashboard {
  background-color: #FFF;
}

.appBarEditor {
  background-color: #EFEFEF;
}

.breadCrumb {
  padding-left: 30;
  font-size: 20;
}

.breadCrumbSubApp {
  color: '#0062BA';
}

.paper {
  text-align: 'center';
}

.subHeader {
  padding-left: 12;
  padding-right: 12;
  padding-bottom: 0;
  font-size: 20;
}

.subHeaderLink {
  border-bottom: '1px solid';
  border-bottom-color: 'transparent';
}
.subHeaderLink:hover {
  border-bottom: '1px solid';
  border-bottom-color: #0062BA;
  color: #0062BA;
}

.subHeaderLinkActive {
  border-bottom: '1px solid';
  border-bottom-color: #0062BA;
  color: #0062BA;
  font-weight: 500;
}
.subHeaderLinkActive:hover {
  border-bottom: '1px solid';
  border-bottom-color: #0062BA;
  color: #0062BA;
}

.topRightService {
  padding-right: 22;
}
