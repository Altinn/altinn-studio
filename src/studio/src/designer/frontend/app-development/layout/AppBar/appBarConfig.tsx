export interface TopBarMenuItem {
  key: TopBarMenu;
  link: string;
}

export enum TopBarMenu {
  About = 'top_menu.about',
  Create = 'top_menu.create',
  Datamodel = 'top_menu.datamodel',
  Text = 'top_menu.texts',
  Deploy = 'top_menu.deploy',
  None = '',
}

export const menu: TopBarMenuItem[] = [
  {
    key: TopBarMenu.About,
    link: '/:org/:app',
  },
  {
    key: TopBarMenu.Create,
    link: '/:org/:app/ui-editor',
  },
  {
    key: TopBarMenu.Datamodel,
    link: '/:org/:app/datamodel',
  },
  {
    key: TopBarMenu.Text,
    link: '/:org/:app/texts',
  },
  {
    key: TopBarMenu.Deploy,
    link: '/:org/:app/deploy',
  },
];
