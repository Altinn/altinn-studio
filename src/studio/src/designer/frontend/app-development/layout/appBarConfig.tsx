export const menu = [
  {
    key: 'Om',
    activeSubHeaderSelection: 'Om',
    link: '/',
  },
  {
    key: 'Lage',
    activeSubHeaderSelection: 'Lage',
    link: '/ui-editor',
  },
  {
    key: 'Språk',
    activeSubHeaderSelection: 'Språk',
    link: '/texts',
  },
  {
    key: 'Deploy',
    activeSubHeaderSelection: 'Deploy',
    link: '/deploy',
  },
];
