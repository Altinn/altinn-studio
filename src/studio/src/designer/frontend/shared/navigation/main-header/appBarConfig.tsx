export const menu = [
  {
    key: 'Om',
    activeSubHeaderSelection: 'Om',
    link: '/about',
  },
  {
    key: 'Lage',
    activeSubHeaderSelection: 'Lage',
    link: '/edit',
  },
  {
    key: 'Språk',
    activeSubHeaderSelection: 'Språk',
    link: '/texts',
  },
  {
    key: 'Oversett',
    activeSubHeaderSelection: 'Oversett1',
    link: '/language-editor',
  },
  {
    key: 'Deploy',
    activeSubHeaderSelection: 'Deploy',
    link: '/deploy',
  },
];
