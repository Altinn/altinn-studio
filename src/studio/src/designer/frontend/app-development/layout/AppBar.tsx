import React, { useState } from 'react';
import type { Theme } from '@mui/material';
import {
  AppBar as MuiAppBar,
  Grid,
  Toolbar,
  useMediaQuery,
} from '@mui/material';
import { createStyles, makeStyles } from '@mui/styles';
import classNames from 'classnames';
import { Link, useParams } from 'react-router-dom';
import { altinnImgLogoHeaderUrl } from 'app-shared/utils/urlHelper';
import type { IMenuItem } from 'app-shared/navigation/drawer/drawerMenuSettings';
import TabletDrawerMenu from 'app-shared/navigation/drawer/TabletDrawerMenu';
import { menu } from './appBarConfig';
import ProfileMenu from 'app-shared/navigation/main-header/profileMenu';
import { useAppSelector } from 'common/hooks';
import { getLanguageFromKey } from 'app-shared/utils/language';

export interface IAppBarProps {
  activeSubHeaderSelection?: string;
  activeLeftMenuSelection?: string;
  logoutButton?: boolean;
  user?: string;
  showSubMenu?: boolean;
  mainMenuItems?: IMenuItem[];
  subMenuItems?: { [key: string]: IMenuItem[] };
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      zIndex: 1,
      color: theme.altinnPalette.primary.black,
    },
    appBar: {
      borderBottom: '1px solid',
      borderBottomColor: '#C9C9C9',
      color: theme.altinnPalette.primary.black,
      position: 'fixed',
    },
    appBarDashboard: {
      backgroundColor: theme.altinnPalette.primary.white,
    },
    appBarEditor: {
      backgroundColor: theme.altinnPalette.primary.greyLight,
    },
    breadCrumb: {
      paddingLeft: 30,
      fontSize: 20,
    },
    breadCrumbSubApp: {
      color: '#0062BA',
    },
    paper: {
      textAlign: 'center',
    },
    subHeader: {
      paddingLeft: 12,
      paddingRight: 12,
      paddingBottom: 0,
      fontSize: 20,
    },
    subHeaderLink: {
      borderBottom: '1px solid',
      borderBottomColor: 'transparent', // To mitigate the 1 pixel adjustment
      '&:hover': {
        borderBottom: '1px solid',
        borderBottomColor: theme.altinnPalette.primary.blueDark,
        color: theme.altinnPalette.primary.blueDark,
      },
    },
    subHeaderLinkActive: {
      borderBottom: '1px solid',
      borderBottomColor: theme.altinnPalette.primary.blueDark,
      color: theme.altinnPalette.primary.blueDark,
      fontWeight: 500,
      '&:hover': {
        borderBottom: '1px solid',
        borderBottomColor: theme.altinnPalette.primary.blueDark,
        color: theme.altinnPalette.primary.blueDark,
      },
    },
    topRightService: {
      paddingRight: 22,
    },
    aImgStyling: {
      borderBottom: 'none',
      '&:hover': {
        borderBottom: 'none',
      },
      '&:active': {
        borderBottom: 'none',
      },
    },
  }),
);

export const AppBar = ({
  activeLeftMenuSelection,
  activeSubHeaderSelection,
  logoutButton,
  user,
  mainMenuItems,
  subMenuItems,
  showSubMenu,
}: IAppBarProps) => {
  const classes = useStyles();
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const hiddenMdUp = useMediaQuery((theme: Theme) =>
    theme.breakpoints.up('md'),
  );
  const hiddenSmDown = useMediaQuery((theme: Theme) =>
    theme.breakpoints.down('sm'),
  );

  const language = useAppSelector((state) => state.languageState.language);
  const t = (key: string) => getLanguageFromKey(key, language);

  const handleDrawerMenuClick = () => {
    setIsMenuOpen((prev) => !prev);
  };
  const { org, app } = useParams();
  return (
    <div className={classes.root}>
      <MuiAppBar
        position='fixed'
        className={classNames([
          !app ? classes.appBarDashboard : classes.appBarEditor,
          classes.appBar,
        ])}
        elevation={5}
        sx={{
          backgroundColor: '#EFEFEF',
          boxShadow: 'none',
          color: 'black',
        }}
      >
        <Toolbar>
          <Grid
            container
            direction='row'
            alignItems='center'
            justifyContent='space-between'
          >
            <Grid
              item
              xs
              container
            >
              <Grid item>
                <a
                  href='/'
                  className={classes.aImgStyling}
                >
                  <img
                    src={altinnImgLogoHeaderUrl}
                    alt='Altinn logo'
                  />
                </a>
              </Grid>
              {hiddenMdUp ? null : (
                <Grid
                  item
                  className={classes.breadCrumb}
                >
                  {activeSubHeaderSelection &&
                    `/ ${t(activeSubHeaderSelection)}`}{' '}
                  /
                  <span className={classes.breadCrumbSubApp}>
                    {' '}
                    {activeLeftMenuSelection}{' '}
                  </span>
                </Grid>
              )}
            </Grid>
            {hiddenSmDown ? null : (
              <Grid
                xs
                item
                className={classes.paper}
              >
                {org && app ? `${org} / ${app}` : ''}
              </Grid>
            )}
            <Grid
              item
              xs
              container
              direction='row'
              alignItems='center'
              justifyContent='flex-end'
            >
              {user || ''}
              {hiddenSmDown ? null : (
                <Grid item>
                  <ProfileMenu showlogout />
                </Grid>
              )}
              {hiddenMdUp ? null : (
                <Grid item>
                  <TabletDrawerMenu
                    handleTabletDrawerMenu={handleDrawerMenuClick}
                    tabletDrawerOpen={isMenuOpen}
                    logoutButton={!logoutButton ? false : logoutButton}
                    activeSubHeaderSelection={activeSubHeaderSelection}
                    activeLeftMenuSelection={activeLeftMenuSelection}
                    mainMenuItems={mainMenuItems}
                    leftDrawerMenuItems={subMenuItems}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Toolbar>
        {hiddenSmDown
          ? null
          : showSubMenu && (
              <Toolbar>
                <Grid
                  container
                  direction='row'
                  justifyContent='center'
                  alignItems='center'
                >
                  {menu.map((item) => (
                    <Grid
                      item
                      key={item.key}
                      className={classNames(classes.subHeader)}
                    >
                      <Link
                        to={item.link.replace(':org', org).replace(':app', app)}
                        className={classNames(classes.subHeaderLink, {
                          [classes.subHeaderLinkActive]:
                            activeSubHeaderSelection === item.key,
                        })}
                      >
                        {t(item.key)}
                      </Link>
                    </Grid>
                  ))}
                </Grid>
              </Toolbar>
            )}
      </MuiAppBar>
    </div>
  );
};
