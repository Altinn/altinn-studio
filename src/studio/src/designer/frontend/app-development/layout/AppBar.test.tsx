import React from 'react';
import { createTheme, ThemeProvider } from '@mui/material';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render as rtlRender, screen } from '@testing-library/react';
import type { IAppBarProps } from './AppBar';
import { AppBar } from './AppBar';
import altinnTheme from 'app-shared/theme/altinnStudioTheme';
import { menu } from './appBarConfig';

describe('AppBar', () => {
  describe('When using AppBarConfig menu entries', () => {
    menu.forEach((entry) => {
      it(`should render ${entry.key} as current item when activeSubHeaderSelection is set to ${entry.key}`, () => {
        const { container } = render({
          activeSubHeaderSelection: entry.key,
          showSubMenu: true,
        });

        const activeClassNamePartial = 'subHeaderLinkActive';

        expect(
          container.querySelectorAll(`[class*="${activeClassNamePartial}"]`),
        ).toHaveLength(1);

        const link = screen.getByRole('link', { name: entry.key });
        const hasActiveLinkClass = Array.from(link.classList).some((x) =>
          x.includes(activeClassNamePartial),
        );

        expect(hasActiveLinkClass).toBe(true);
      });
    });
  });
});

const render = (props: Partial<IAppBarProps> = {}) => {
  const themeCopy = { ...altinnTheme };
  themeCopy.props.MuiWithWidth = { initialWidth: 'md' };
  const theme = createTheme(themeCopy);
  const allProps = {
    org: 'jest-test-org',
    app: 'jest-test-app',
    showSubMenu: true,
    activeSubHeaderSelection: 'Lage',
    ...props,
  } as IAppBarProps;

  const createStore = configureStore();
  const initialState = {
    languageState: {
      language: {},
    },
  };
  const store = createStore(initialState);

  return rtlRender(
    <MemoryRouter>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <AppBar {...allProps} />
        </ThemeProvider>
      </Provider>
    </MemoryRouter>,
  );
};
