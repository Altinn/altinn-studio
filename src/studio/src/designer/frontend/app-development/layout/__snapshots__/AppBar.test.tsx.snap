// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Snapshot should match snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-2 makeStyles-appBarEditor-4 MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="makeStyles-aImgStyling-12"
              href="/"
            >
              <img
                alt="Altinn logo"
                src="https://altinncdn.no/img/Altinn-logo-blue.svg"
              />
            </a>
          </div>
          <div
            class="MuiGrid-root makeStyles-breadCrumb-5 MuiGrid-item"
          >
            / Lage
             
            /
            <span
              class="makeStyles-breadCrumbSubApp-6"
            >
               
               
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-paper-7 MuiGrid-item MuiGrid-grid-xs-true"
        >
          jest-test-org / jest-test-app
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="profilikon knapp"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-label="profilikon"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <title>
                      profilikon
                    </title>
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-15"
            >
              <div
                class="MuiPaper-root makeStyles-drawerMenuPaper-21 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
              >
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary makeStyles-menuItemText-19 MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Logg ut
                      </span>
                    </div>
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root makeStyles-divider-16"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root makeStyles-mainMenuItem-20 MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="MuiListItemText-root makeStyles-mainMenuItem-20"
                      >
                        Språk
                      </div>
                    </div>
                    <hr
                      class="MuiDivider-root makeStyles-divider-16"
                    />
                  </div>
                </ul>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-commonButton-22 makeStyles-closeButton-24 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                meny
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-subHeader-8 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-9"
            href="/"
          >
            Om
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-8 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-9 makeStyles-subHeaderLinkActive-10"
            href="/ui-editor"
          >
            Lage
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-8 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-9"
            href="/texts"
          >
            Språk
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-8 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-9"
            href="/deploy"
          >
            Deploy
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`AppBar Snapshot should match snapshot and not render subHeader menu 1`] = `
<div
  class="makeStyles-root-51"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-52 makeStyles-appBarEditor-54 MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="makeStyles-aImgStyling-62"
              href="/"
            >
              <img
                alt="Altinn logo"
                src="https://altinncdn.no/img/Altinn-logo-blue.svg"
              />
            </a>
          </div>
          <div
            class="MuiGrid-root makeStyles-breadCrumb-55 MuiGrid-item"
          >
            / Lage
             
            /
            <span
              class="makeStyles-breadCrumbSubApp-56"
            >
               
               
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-paper-57 MuiGrid-item MuiGrid-grid-xs-true"
        >
          jest-test-org / jest-test-app
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="profilikon knapp"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-label="profilikon"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <title>
                      profilikon
                    </title>
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-65"
            >
              <div
                class="MuiPaper-root makeStyles-drawerMenuPaper-71 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
              >
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary makeStyles-menuItemText-69 MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Logg ut
                      </span>
                    </div>
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root makeStyles-divider-66"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root makeStyles-mainMenuItem-70 MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="MuiListItemText-root makeStyles-mainMenuItem-70"
                      >
                        Språk
                      </div>
                    </div>
                    <hr
                      class="MuiDivider-root makeStyles-divider-66"
                    />
                  </div>
                </ul>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-commonButton-72 makeStyles-closeButton-74 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                meny
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`AppBar Snapshot should match snapshot with no app or org 1`] = `
<div
  class="makeStyles-root-76"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-77 makeStyles-appBarDashboard-78 MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="makeStyles-aImgStyling-87"
              href="/"
            >
              <img
                alt="Altinn logo"
                src="https://altinncdn.no/img/Altinn-logo-blue.svg"
              />
            </a>
          </div>
          <div
            class="MuiGrid-root makeStyles-breadCrumb-80 MuiGrid-item"
          >
             
            /
            <span
              class="makeStyles-breadCrumbSubApp-81"
            >
               
               
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-paper-82 MuiGrid-item MuiGrid-grid-xs-true"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="profilikon knapp"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-label="profilikon"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <title>
                      profilikon
                    </title>
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-90"
            >
              <div
                class="MuiPaper-root makeStyles-drawerMenuPaper-96 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
              >
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary makeStyles-menuItemText-94 MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Logg ut
                      </span>
                    </div>
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root makeStyles-divider-91"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root makeStyles-mainMenuItem-95 MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="MuiListItemText-root makeStyles-mainMenuItem-95"
                      >
                        Språk
                      </div>
                    </div>
                    <hr
                      class="MuiDivider-root makeStyles-divider-91"
                    />
                  </div>
                </ul>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-commonButton-97 makeStyles-closeButton-99 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                meny
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`AppBar Snapshot should match snapshot with subHeader and Publisere selection active 1`] = `
<div
  class="makeStyles-root-26"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-27 makeStyles-appBarEditor-29 MuiPaper-elevation0"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="makeStyles-aImgStyling-37"
              href="/"
            >
              <img
                alt="Altinn logo"
                src="https://altinncdn.no/img/Altinn-logo-blue.svg"
              />
            </a>
          </div>
          <div
            class="MuiGrid-root makeStyles-breadCrumb-30 MuiGrid-item"
          >
            / Publisere
             
            /
            <span
              class="makeStyles-breadCrumbSubApp-31"
            >
               
               
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-paper-32 MuiGrid-item MuiGrid-grid-xs-true"
        >
          other-org / other-app
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-true"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="profilikon knapp"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-label="profilikon"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                    <title>
                      profilikon
                    </title>
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-40"
            >
              <div
                class="MuiPaper-root makeStyles-drawerMenuPaper-46 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
              >
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiListItemText-primary makeStyles-menuItemText-44 MuiTypography-body1 MuiTypography-displayBlock"
                      >
                        Logg ut
                      </span>
                    </div>
                  </div>
                </ul>
                <hr
                  class="MuiDivider-root makeStyles-divider-41"
                />
                <ul
                  class="MuiList-root MuiList-padding"
                >
                  <div>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root makeStyles-mainMenuItem-45 MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="MuiListItemText-root makeStyles-mainMenuItem-45"
                      >
                        Språk
                      </div>
                    </div>
                    <hr
                      class="MuiDivider-root makeStyles-divider-41"
                    />
                  </div>
                </ul>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-commonButton-47 makeStyles-closeButton-49 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                meny
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-subHeader-33 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-34"
            href="/"
          >
            Om
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-33 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-34"
            href="/ui-editor"
          >
            Lage
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-33 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-34"
            href="/texts"
          >
            Språk
          </a>
        </div>
        <div
          class="MuiGrid-root makeStyles-subHeader-33 MuiGrid-item"
        >
          <a
            class="makeStyles-subHeaderLink-34"
            href="/deploy"
          >
            Deploy
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
